run.go

```diff
@@ -13,48 +15,223 @@ import (
 // execution times are reported, as well as using the test cache. Do not import a test package into another test package
 // as that will cause the tests in the second package to get executed with the first package. If code or resources need
 // shared between test packages, put them in their own package which does not contain any test definitions.
+//
+// Individual tests in a package may still be run using the standard -run test flag. See `go help testflag` for more
+// information.
+//
+// TODO: shuffle test execution order (see -shuffle in `go help testflag`)
 func RunAsTest(t *testing.T) {
 	t.Helper()
-	instance.tests.Iterate(func(pkg string, tests orderedmap.OrderedMap[string, testCase]) bool {
-		tests.Iterate(func(name string, test testCase) bool {
-			t.Run(test.Name, func(tt *testing.T) {
-				tt.Helper()
-				test.tester(tWrapper{t: tt})
+	instance.tests.Iterate(func(pkg string, pkgTests *testPkg) bool {
+		// we have to hold onto any panics here to be able to run AfterPackage
```

Comment:
this whole thing is kinda ugly and I can't think of a way to test it since it is supposed to run inside a test anyway.